### PR TITLE
Offer 'field' when within a property accessor.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/EditorFeatures/CSharpTest2/Recommendations/FieldKeywordRecommenderTests.cs [hunk 1]
```diff
@@ -7,222 +7,419 @@
 using Microsoft.CodeAnalysis.Test.Utilities;
 using Xunit;
 
-namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Recommendations
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Recommendations;
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest2/Recommendations/FieldKeywordRecommenderTests.cs:10]: view with whitespace off.

**Review by dibarbet — APPROVED** [review]

File: src/Features/CSharp/Portable/Completion/KeywordRecommenders/FieldKeywordRecommender.cs [hunk 2]
```diff
@@ -20,11 +22,36 @@ internal class FieldKeywordRecommender : AbstractSyntacticSingleKeywordRecommend
...
     {
+        // `[field:` is legal in an attribute within a type.
+        if (context.IsMemberAttributeContext(s_validTypeDeclarations, cancellationToken))
+            return true;
+
+        // Check if we're within a property accessor where the `field` keyword is legal.  Note: we do not do a lang
+        // version check here.  We do not want to interfere with users trying to use/learn this feature.  The user will
+        // get a clear message if they're not on the right lang version telling them about the issue, and offering to
+        // upgrade their project if they way.
+        if (context.IsAnyExpressionContext || context.IsStatementContext)
```
Comment (sharwell) [inline src/Features/CSharp/Portable/Completion/KeywordRecommenders/FieldKeywordRecommender.cs:35]: 💭 Is there a case where `IsAnyExpressionContext` is false but `IsStatementContext` is true?

File: src/Features/CSharp/Portable/Completion/KeywordRecommenders/FieldKeywordRecommender.cs [hunk 3]
```diff
@@ -20,11 +22,36 @@ internal class FieldKeywordRecommender : AbstractSyntacticSingleKeywordRecommend
...
+        }
+
+        return false;
     }
 
-    protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
-        => context.IsMemberAttributeContext(s_validTypeDeclarations, cancellationToken);
+    private static bool IsInPropertyAccessor(SyntaxToken targetToken)
+    {
+        for (var node = targetToken.Parent; node != null; node = node.Parent)
```
Comment (sharwell) [inline src/Features/CSharp/Portable/Completion/KeywordRecommenders/FieldKeywordRecommender.cs:46]: ❔ Does this intentionally walk out of local functions? Even if so, it seems like static local functions should be skipped.

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/Completion/KeywordRecommenders/FieldKeywordRecommender.cs:46]: yes.  it intentionally walks out of local functions.

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/Completion/KeywordRecommenders/FieldKeywordRecommender.cs:46]: >  Even if so, it seems like static local functions should be skipped.

`field` is allowed ina  static local/lambda if hte containing property is also static.  We could consdier adding the logic to check for that.  but i'm in the not-caring poitn currently.

**Pull request merged**